### DISCUSSION TITLE
Cycle Routing following Named/Numbered Bicycle Route

### DISCUSSION THREAD

**Discussion by AndrewSnow** [osmandapp/OsmAnd#12134] [Q&A]
In many parts of the world long routes intended for use by touring bicyclists have been developed, given names or route numbers and have been documented in OSM (see the CyclOSM or Cycle Map map layers).  In the U.S., these exist in many states as part of the numbered US Bicycle Route System (USBRS - see https://en.wikipedia.org/wiki/United_States_Bicycle_Route_System) and represents bicycle safe routes hundreds and even thousands of miles in length (see examples in the wiki page).  Europe has a similar bike route network: https://en.wikipedia.org/wiki/EuroVelo  In OSMAnd however, I can't seem to find a way to tell the app I want to follow one of these routes.  In Route Parameters, we are given the option to for example "Avoid Motorways" but do not have the option to "Prefer Designated Bike Route".  Am I missing a feature or setting?

**Comment by vshcherb** [comment]
Today it's not possible but you can enable it and create segment by segment with Plan Route functionality.

**Comment by scaidermern** [comment]
I think this is related to #7299 which asks to make routes selectable and routable.

**Comment by Themanwithoutaplan** [comment]
FWIW I think this feature has been available for a while but its utility is severely limited, especially on longer routes. For example, I'm planning to cycle along part the Baltic Sea bike route, which runs from the Danish border along the German coast to Poland. In total this is around 780 km, it's directly accessible in Osmand and can be downloaded, but it's virtually unusable for going between points on the route. If you enable it in navigation it will load the whole route, even if you have already set start and finish points – you can add these back in later, but it's very confusing!

Cycling along the Loire last year was a little more successful because the track is broken down into sections Osmand was quite happy to "snap" to the track if you're close to it, but planning in advance is much more difficult.

I wonder if I'm missing something because there is a navigation option "Follow track", but this replaces the starting point, with the start of the track, which could be hundreds of kilometres away. You can replace this with the previous starting point, and add the destination, but this means you have to add markers and can't just navigate ad hoc.

Is there any better way of handling this other than chopping the track into section in another app – Osmand is really not suitable for this?

**Comment by wilfriedh** [comment]
I understand the OP's wish. 
In principle it is possible to implement a routing algorithm which prefers the numbered routes. Outdooractive does so. 
However, (in Germany where I cycle) following the official numbered route has pros and cons. 
My experiences: 
1. the numbered route is sometimes much longer than necessary, even where OsmAnd's route uses secondary, low traffic ways.
2. sometimes the official numbered route follows a street with high motorized traffic, while OsmAnd suggests a parallel low traffic street. Or the official numbered route includes a steep climb and descent, while OsmAnd (with the option to account for level data, which I do like very much, many thanks for that!) suggests a very nice alternative route. 

When I plan a trip, I always look both into the route suggestion on OsmAnd and of Outdooractive, and sometimes in google maps satellite view, and what the mapped ways look like, and then decide which ways to choose. For me, none of the routing algorithms is perfect in all cases.

**Reply by Themanwithoutaplan** [reply]
Sure. But, it should be possible to make it easier to use tracks in the current setup. If you're already on the route then adding a destination and following the track works pretty much as you'd expect, and I've done this a lot. But if you're not and then it's very frustrating to possibly add a starting point and destination and then choose a track, only to find you have to start again. The routing should be from the start to the track, follow the track to the point closest to the destination.
